lib/tapioca/compilers/dsl/active_record_columns.rb

```diff
@@ -0,0 +1,347 @@
+      # # db/schema.rb
+      # create_table :posts do |t|
+      #   t.string :title, null: false
+      #   t.string :body
+      #   t.boolean :published
+      #   t.timestamps
+      # end
+      # ~~~
+      #
+      # this generator will produce the following methods in the RBI file
+      # `post.rbi`:
+      #
+      # ~~~rbi
+      # # post.rbi
+      # # typed: true
+      # class Post
+      #   sig { returns(T.nilable(::String)) }
+      #   def body; end
+      #
+      #   sig { params(value: ::String).returns(::String) }
+      #   def body=; end
+      #
+      #   sig { params(args: T.untyped).returns(T::Boolean) }
+      #   def body?; end
+      #
+      #   sig { returns(T.nilable(::ActiveSupport::TimeWithZone)) }
+      #   def created_at; end
+      #
+      #   sig { params(value: ::ActiveSupport::TimeWithZone).returns(::ActiveSupport::TimeWithZone) }
+      #   def created_at=; end
+      #
+      #   sig { params(args: T.untyped).returns(T::Boolean) }
+      #   def created_at?; end
+      #
+      #   sig { returns(T.nilable(T::Boolean)) }
+      #   def published; end
+      #
+      #   sig { params(value: T::Boolean).returns(T::Boolean) }
+      #   def published=; end
+      #
+      #   sig { params(args: T.untyped).returns(T::Boolean) }
+      #   def published?; end
+      #
+      #   sig { returns(::String) }
+      #   def title; end
+      #
+      #   sig { params(value: ::String).returns(::String) }
+      #   def title=(value); end
+      #
+      #   sig { params(args: T.untyped).returns(T::Boolean) }
+      #   def title?(*args); end
+      #
+      #   sig { returns(T.nilable(::ActiveSupport::TimeWithZone)) }
+      #   def updated_at; end
+      #
+      #   sig { params(value: ::ActiveSupport::TimeWithZone).returns(::ActiveSupport::TimeWithZone) }
+      #   def updated_at=; end
+      #
+      #   sig { params(args: T.untyped).returns(T::Boolean) }
+      #   def updated_at?; end
+      #
+      #   ## Also the methods added by https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Dirty.html
+      #   ## Also the methods added by https://api.rubyonrails.org/classes/ActiveModel/Dirty.html
+      #   ## Also the methods added by https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/BeforeTypeCast.html
+      # end
+      # ~~~
+      class ActiveRecordColumns < Base
+        extend T::Sig
+
+        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: T.class_of(ActiveRecord::Base)).void }
+        def decorate(root, constant)
+          return unless constant.table_exists?
+
+          module_name = "#{constant}::GeneratedAttributeMethods"
+          root.create_module(module_name) do |mod|
+            constant.columns_hash.each_key do |column_name|
+              column_name = column_name.to_s
+              add_methods_for_attribute(mod, constant, column_name)
+            end
+
+            constant.attribute_aliases.each do |new_name, old_name|
```

Comment:
We need to test aliased attributes as well. We also need to test aliased attributes when the aliased attribute ends up conflicting with one of the generated attribute methods. For example:

```ruby
alias_attribute :nickname, :name
alias_attribute :name?, :name # yes, people do stuff like this
```

That's the point of the `old_method_names`/`new_method_names` shenanigans below.